var/spack/repos/builtin/packages/cbtf-krell/package.py

```diff
@@ -263,3 +267,49 @@ def cmake_args(self):
             self.set_cray_login_node_cmake_options(spec, cmake_args)
 
         return cmake_args
+
+    def setup_environment(self, spack_env, run_env):
+        """Set up the compile and runtime environments for a package."""
+
+        # Environment settings for cbtf-krell
+        run_env.prepend_path('PATH', self.prefix.bin)
+        run_env.prepend_path('PATH', self.prefix.sbin)
+
+        # Find openspeedshop library path
+        cbtfk_libdir = find_libraries('libcbtf-core.so.1.1.0',
+                                      root=self.prefix,
+                                      shared=True, recursive=True)
+        run_env.prepend_path('LD_LIBRARY_PATH',
+                             os.path.dirname(cbtfk_libdir.joined()))
+
+        cbtf_mc = '/sbin/cbtf_mrnet_commnode'
+        cbtf_lmb = '/sbin/cbtf_libcbtf_mrnet_backend'
+        run_env.set('XPLAT_RSH', 'ssh')
+        run_env.set('MRNET_COMM_PATH',
+                    join_path(self.spec.prefix + cbtf_mc))
+
+        # Set CBTF_MPI_IMPLEMENTATON to the appropriate mpi implementation
+        # This is needed by CBTF tools to deploy the correct
+        # mpi runtimes for cbtfsummary
+        # Users may have to set the CBTF_MPI_IMPLEMENTATION variable
+        # manually if multiple mpi's are specified in the build
+
+        if self.spec.satisfies('+mpich'):
+            run_env.set('CBTF_MPI_IMPLEMENTATION', "mpich")
+
+        if self.spec.satisfies('+mvapich'):
+            run_env.set('CBTF_MPI_IMPLEMENTATION', "mvapich")
+
+        if self.spec.satisfies('+mvapich2'):
+            run_env.set('CBTF_MPI_IMPLEMENTATION', "mvapich2")
+
+        if self.spec.satisfies('+mpt'):
+            run_env.set('CBTF_MPI_IMPLEMENTATION', "mpt")
+
+        if self.spec.satisfies('+openmpi'):
+            run_env.set('CBTF_MPI_IMPLEMENTATION', "openmpi")
+
+        run_env.set('CBTF_MRNET_BACKEND_PATH',
+                    join_path(self.prefix + cbtf_lmb))
+        run_env.prepend_path('PATH', self.spec['mrnet'].prefix.bin)
+        run_env.prepend_path('PATH', self.spec['python'].prefix.bin)
```

Comment:
There's been a lot of debate as to whether packages should manually add their runtime dependencies to the `PATH`. The consensus in #8639 seems to be that they shouldn't, but that Spack should automatically load the modules for those runtime dependencies by default.

I would leave out this change, but make sure that `mrnet` and `python` are `type=('build', 'link', 'run')` and `type=('build', 'run')`, respectively. For now, you should tell people to configure Spack to autoload run deps. In the long run, Spack should do this automatically.

Reply:
@adamjstewart Thanks for all your help!   It is much appreciated!
I've been running into an issue on a power 9 cluster where openspeedshop can't find a cbtf-krell library.  I wonder if it is because I'm not tagging the cbtf-krell as "run, link, build" on the depends_on(cbtf-krell) line in openspeedshop/package.py.   I'm going to change my depends_on statements to be more explicit.

But, how does one configure spack to autoload run deps?

Reply:
https://spack.readthedocs.io/en/latest/module_file_support.html#autoload-dependencies